math/matrix/printmatrix.go

```diff
@@ -0,0 +1,30 @@
+package matrix
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// PrintMatrix prints a 2D matrix of any type.
+func PrintMatrix(matrix interface{}) {
```

Comment:
Use the `any` keyword instead of `interface{}`. Also, `[][]any` would allow you to skip those type checks.